src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,255 @@
+###
+# Computing (tropical) Groebner bases in Oscar
+# ============================================
+#
+# For a definition of tropical Groebner basis see Section 2.4 in:
+#   D. Maclagan, B. Sturmfels: Introduction to tropical geometry
+# To see how they can be computed using standard bases see:
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly{K},fmpq_mpoly,fmpz_mpoly} where {K})
+  leadexpv,tailexpvs = Iterators.peel(exponent_vectors(f))
+  d = sum(leadexpv)
+  for tailexpv in tailexpvs
+    if d!=sum(tailexpv)
+      return false
+    end
+  end
+  return true
+end
+export is_homogeneous
+
+function is_homogeneous(I::MPolyIdeal{K} where {K})
+  # todo: test whether generators are interreduced
+  @warn "is_homogeneous: merely checking whether given generators are homogeneous, can result in false negative"
+
+  for f in gens(I)
+    if !is_homogeneous(f)
+      return false
+    end
+  end
+  return true
```

Comment:
More idiomatic would be:
```suggestion
  return all(is_homogeneous, gens(I))
```